### PR TITLE
chore: Add GitHub action for build and test

### PR DESCRIPTION
**Pull request by moritzploss-k**
## About

Add a GitHub action to build and test the project.

### CODE REVIEW

File: src/jsonformat.app.src [hunk 1]
```diff
@@ -15,7 +15,7 @@
 {application, jsonformat,
   [ {description,  "A custom formatter for the erlang logger applications that "
                    "outputs JSON formatted single line logs"}
-  , {vsn,           "1.0.1"}
+  , {vsn,           "0.0.1"}
```
Comment (moritzploss-k) [inline src/jsonformat.app.src:18]: just resetting to default since we don't usually change this

Comment (jakobsvenning) [reply src/jsonformat.app.src:18]: Don't we normally put `vsn` to the atom `git`?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    - name: Compile
+      run: make compile
+    - name: Run xref
+      run: make xref
+    - name: Run dialyzer
+      run: make dialyze
+    - name: Run eunit tests
+      run: make eunit
+    - name: Run ct tests
+      run: make ct
```
Comment (jakobsvenning) [inline .github/workflows/build.yml:26]:
```suggestion
```

We don't have any CT tests in this library but I guess it does not hurt to still run the command..

**Review by jakobsvenning — APPROVED** [review]

File: src/jsonformat.app.src (hunk 1 above)
Comment (moritzploss-k) [reply src/jsonformat.app.src:18]: good point!

File: .github/workflows/build.yml (hunk 2 above)
Comment (moritzploss-k) [reply .github/workflows/build.yml:26]: that's what I thought

**Pull request merged**